frontend/src/services/api.ts

```diff
@@ -0,0 +1,33 @@
+export type RequestState<T> = Ok<T> | Error | Loading;
+
+interface Ok<T> {
+  type: "ok";
+  routes: T;
```

Comment:
routes should be an extended type I think? There could be other Ok requests that don't have the routes property

Reply:
Whoops, I moved it out of RouteList to make it reusable elsewhere and forgot the member var...